anxcloud/utils.go

```diff
@@ -99,3 +102,13 @@ func mustCastInterfaceArray[T any](in []interface{}) []T {
 	}
 	return out
 }
+
+var lowercaseHostnameRFC1123RegExp = regexp.MustCompile(`^([a-z0-9]{1}[a-z0-9-]{0,62}){1}(\.[a-z0-9]{1}[a-z0-9-]{0,62})*?$`)
```

Comment:
if we keep this, please take the regex from our internal backend (GS definition for the Cluster resource)